newrelic/hooks/mlmodel_openai.py

```diff
@@ -1280,6 +1329,10 @@ def wrap_stream_iter_events_async(wrapped, instance, args, kwargs):
     if not transaction:
         return wrapped(*args, **kwargs)
 
+    settings = transaction.settings if transaction.settings is not None else global_settings()
+    if not settings.ai_monitoring.enabled:
+        return wrapped(*args, **kwargs)
+
     proxied_return_val = AsyncGeneratorProxy(wrapped(*args, **kwargs))
```

Comment:
We also need to check this ai_monitoring setting inside the following wrappers:
* wrap_base_client_process_response_sync
* wrap_base_client_process_response_async
* wrap_convert_to_openai_object
* wrap_stream_iter_events_sync
* wrap_stream_iter_events_async
* wrap_engine_api_resource_create_async
* wrap_engine_api_resource_create_sync